### PR TITLE
feat: `std` feature flag for `alloy-consensus`

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Ports `alloy-consensus` to be a `no_std` compatible crate.

### CODE REVIEW

File: crates/consensus/src/transaction/eip4844/builder.rs [hunk 1]
```diff
@@ -3,6 +3,14 @@ use alloy_eips::eip4844::Blob;
 #[cfg(feature = "kzg")]
 use c_kzg::{Blob, KzgCommitment, KzgProof};
 
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+#[cfg(not(feature = "std"))]
+use core::cmp;
```
Comment (prestwich) [inline crates/consensus/src/transaction/eip4844/builder.rs:9]: no need to feature gate core ever as it is part of both the `std` prelude and the `core` prelude.

**Review by prestwich — CHANGES_REQUESTED** [review]
meeds update to the no_std list in the readme

File: crates/consensus/src/transaction/eip4844/builder.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/consensus/src/transaction/eip4844/builder.rs:9]: Yes please just use `core` and `alloc` instead of feature gating imports

**Comment by DaniPopes** [comment]
There's also `scripts/check_no_std.sh`

File: crates/consensus/src/transaction/eip4844/builder.rs (hunk 1 above)
Comment (prestwich) [reply crates/consensus/src/transaction/eip4844/builder.rs:9]: alloc needs to be feature gated due to new redundant import lints

File: crates/consensus/src/receipt/mod.rs [hunk 2]
```diff
@@ -37,6 +37,7 @@ mod tests {
     use alloy_eips::eip2718::Encodable2718;
     use alloy_primitives::{address, b256, bytes, hex, Bytes, LogData};
     use alloy_rlp::{Decodable, Encodable};
+    use std::vec;
```
Comment (prestwich) [inline crates/consensus/src/receipt/mod.rs:40]: this seems out of place. does it not cause redundant import lints?

Comment (clabby) [reply crates/consensus/src/receipt/mod.rs]: Ah good spot - not necessary, removed.

**Review by prestwich — APPROVED** [review]

**Comment by prestwich** [comment]
rebased onto main after #454 to re-run CI with new changes. shouldn't be any conflicts, but 454 was big. will merge after CI passes

**Pull request merged**